elide-graphql/src/main/java/com/yahoo/elide/graphql/GraphQLEndpoint.java

```diff
@@ -64,20 +68,43 @@ public GraphQLEndpoint(@Named("elide") Elide elide) {
     @Consumes(MediaType.APPLICATION_JSON)
     public Response post(
             @Context UriInfo uriInfo,
-            @HeaderParam("ApiVersion") String apiVersion,
+            @Context HttpHeaders headers,
             @Context SecurityContext securityContext,
             String graphQLDocument) {
 
-        String safeApiVersion = apiVersion == null ? NO_VERSION : apiVersion;
+        String apiVersion = resolveApiVersion(headers);
+        MultivaluedMap<String, String> requestHeaders = removeAuthHeaders(headers);
         User user = new SecurityContextUser(securityContext);
-        QueryRunner runner = runners.getOrDefault(safeApiVersion, null);
+        QueryRunner runner = runners.getOrDefault(apiVersion, null);
 
         ElideResponse response;
         if (runner == null) {
             response = buildErrorResponse(elide, new InvalidOperationException("Invalid API Version"), false);
         } else {
-            response = runner.run(uriInfo.getBaseUri().toString(), graphQLDocument, user);
+            response = runner.run(uriInfo.getBaseUri().toString(),
+                                  graphQLDocument, user, UUID.randomUUID(), requestHeaders);
         }
         return Response.status(response.getResponseCode()).entity(response.getBody()).build();
     }
+
+    private String resolveApiVersion(HttpHeaders headers) {
```

Comment:
Rather than redefine these methods twice, let's put them in a common class (HeaderUtils) or something like that.